pkg/kv/dynamodb/store.go

```diff
@@ -0,0 +1,361 @@
+package dynamodb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	svc    *dynamodb.DynamoDB
+	params *Params
+}
+
+type EntriesIterator struct {
+	entry        *kv.Entry
+	err          error
+	store        *Store
+	queryResult  *dynamodb.QueryOutput
+	currEntryIdx int
+	partKey      string
+	startKey     string
+}
+
+type DynKVItem struct {
+	PartitionKey string
+	ItemKey      string
+	ItemValue    string
+}
+
+type Params struct {
+	TableName          string
+	ReadCapacityUnits  int64
+	WriteCapacityUnits int64
+
+	Endpoint           string
+	AwsRegion          string
+	AwsAccessKeyID     string
+	AwsSecretAccessKey string
+}
+
+const (
+	DriverName       = "dynamodb"
+	DefaultTableName = "kvstore"
```

Comment:
Lets keep the convention as in the postgres implementation and set it to "kv".
Also - does it make sense to move this definition to the store interface and have all implementations use it instead?

Reply:
No can do. DynamoDB requires 3 chars minimum for table names :man_shrugging: 
Maybe we should align the other way around?

Reply:
Sounds reasonable